contracts/tests/test_unruggable_memecoin.cairo

```diff
@@ -1221,4 +1221,250 @@ mod memecoin_internals {
             index += 1;
         };
     }
+    #[test]
+    fn test_set_merkle_root() {
```

Comment:
a 1200+ lines test files is a bit of a mess. imo we should separate tests related to a specific feature in a submodule with only tests related to that feature.